src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_BuildSynchronization.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV1
+{
+    internal partial class DiagnosticIncrementalAnalyzer
+    {
+        public override async Task SynchronizeWithBuildAsync(Project project, ImmutableArray<DiagnosticData> diagnostics)
+        {
+            if (!PreferBuildErrors(project.Solution.Workspace))
+            {
+                // prefer live errors over build errors
+                return;
+            }
+
+            using (var poolObject = SharedPools.Default<HashSet<string>>().GetPooledObject())
+            {
+                var lookup = CreateDiagnosticIdLookup(diagnostics);
+
+                foreach (var stateSet in _stateManger.GetStateSets(project))
+                {
+                    var descriptors = HostAnalyzerManager.GetDiagnosticDescriptors(stateSet.Analyzer);
+                    var liveDiagnostics = ConvertToLiveDiagnostics(lookup, descriptors, poolObject.Object);
+
+                    var projectTextVersion = VersionStamp.Default;
+                    var semanticVersion = await project.GetDependentSemanticVersionAsync(CancellationToken.None).ConfigureAwait(false);
+
+                    var state = stateSet.GetState(StateType.Project);
+                    var existingDiagnostics = await state.TryGetExistingDataAsync(project, CancellationToken.None).ConfigureAwait(false);
+
+                    var mergedDiagnostics = MergeDiagnostics(liveDiagnostics, GetExistingDiagnostics(existingDiagnostics));
+                    await state.PersistAsync(project, new AnalysisData(projectTextVersion, semanticVersion, mergedDiagnostics), CancellationToken.None).ConfigureAwait(false);
+                    RaiseDiagnosticsUpdated(StateType.Project, project.Id, stateSet.Analyzer, new SolutionArgument(project), mergedDiagnostics);
+                }
+            }
+        }
+
+        public override async Task SynchronizeWithBuildAsync(Document document, ImmutableArray<DiagnosticData> diagnostics)
+        {
+            if (!PreferBuildErrors(document.Project.Solution.Workspace))
+            {
+                // prefer live errors over build errors
+                return;
+            }
+
+            using (var poolObject = SharedPools.Default<HashSet<string>>().GetPooledObject())
+            {
+                var lookup = CreateDiagnosticIdLookup(diagnostics);
+
+                foreach (var stateSet in _stateManger.GetStateSets(document.Project))
+                {
+                    var textVersion = VersionStamp.Default;
```

Comment:
we are using VersionStamp.Default so that lightbulb can't use cached information.